### ISSUE TITLE
CLI AssertionError when wandb Logger is not initialized with save_dir

### ISSUE THREAD

**Issue by tshu-w** [Lightning-AI/lightning#14222]
## 🐛 Bug

When run CLI with wandb logger, which is not initialized with `save_dir`, we got AssertionError.

### To Reproduce

```python
import os

from pytorch_lightning.loggers import WandbLogger
from pytorch_lightning.cli import LightningCLI
from pytorch_lightning import Trainer

from pytorch_lightning.demos.boring_classes import DemoModel, BoringDataModule

def run():
    logger = WandbLogger(project="bug_report")
    cli = LightningCLI(
        DemoModel,
        BoringDataModule,
        trainer_defaults={"logger": logger, "max_epochs": 2, "default_root_dir": os.getcwd()},
    )


if __name__ == "__main__":
    run()
```
and run
```console
python bug_report_model.py fit
```
Backtrace:
```
/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/utilities/seed.py:55: UserWarning: No seed found, seed set to 2752189640
  rank_zero_warn(f"No seed found, seed set to {seed}")
Global seed set to 2752189640
GPU available: True (mps), used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py:1791: UserWarning: MPS available but not used. Set `accelerator` and `devices` using `Trainer(accelerator='mps', devices=1)`.
  rank_zero_warn(
/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/configuration_validator.py:115: UserWarning: You passed in a `val_dataloader` but have no `validation_step`. Skipping val loop.
  rank_zero_warn("You passed in a `val_dataloader` but have no `validation_step`. Skipping val loop.")
Traceback (most recent call last):
  File "/Users/wangtianshu/Developer/lightning/examples/pl_bug_report/bug_report_model.py", line 19, in <module>
    run()
  File "/Users/wangtianshu/Developer/lightning/examples/pl_bug_report/bug_report_model.py", line 11, in run
    cli = LightningCLI(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/cli.py", line 350, in __init__
    self._run_subcommand(self.subcommand)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/cli.py", line 626, in _run_subcommand
    fn(**fn_kwargs)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 696, in fit
    self._call_and_handle_interrupt(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 650, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 737, in _fit_impl
    results = self._run(model, ckpt_path=self.ckpt_path)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1107, in _run
    self._call_setup_hook()  # allow user to setup lightning_module in accelerator environment
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1450, in _call_setup_hook
    self._call_callback_hooks("setup", stage=fn)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/trainer/trainer.py", line 1599, in _call_callback_hooks
    fn(self, self.lightning_module, *args, **kwargs)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/pl/lib/python3.9/site-packages/pytorch_lightning/cli.py", line 210, in setup
    assert log_dir is not None
AssertionError
```

### Expected behavior

CLI run correctly.

### Environment

<details>
  <summary>Details</summary>
    * CUDA:
	- GPU:               None
	- available:         False
	- version:           None
* Lightning:
	- pytorch-lightning: 1.7.1
	- torch:             1.12.1
	- torchmetrics:      0.9.3
	- torchtext:         0.13.1
* Packages:
	- absl-py:           1.2.0
	- aiohttp:           3.8.1
	- aiosignal:         1.2.0
	- antlr4-python3-runtime: 4.9.3
	- async-timeout:     4.0.2
	- attrs:             22.1.0
	- cachetools:        5.2.0
	- certifi:           2022.6.15
	- charset-normalizer: 2.1.0
	- click:             8.1.3
	- commonmark:        0.9.1
	- cycler:            0.11.0
	- decorator:         5.1.1
	- docker-pycreds:    0.4.0
	- docstring-parser:  0.14.1
	- fonttools:         4.35.0
	- frozenlist:        1.3.1
	- fsspec:            2022.7.1
	- gcsfs:             2022.7.1
	- gitdb:             4.0.9
	- gitpython:         3.1.27
	- google-api-core:   2.8.2
	- google-auth:       2.10.0
	- google-auth-oauthlib: 0.4.6
	- google-cloud-core: 2.3.2
	- google-cloud-storage: 2.5.0
	- google-crc32c:     1.3.0
	- google-resumable-media: 2.3.3
	- googleapis-common-protos: 1.56.4
	- grpcio:            1.47.0
	- hydra-core:        1.2.0
	- idna:              3.3
	- importlib-metadata: 4.12.0
	- jsonargparse:      4.13.1
	- kiwisolver:        1.4.4
	- markdown:          3.4.1
	- markupsafe:        2.1.1
	- matplotlib:        3.5.3
	- multidict:         6.0.2
	- numpy:             1.23.2
	- oauthlib:          3.2.0
	- omegaconf:         2.2.2
	- packaging:         21.3
	- pathtools:         0.1.2
	- pillow:            9.2.0
	- pip:               22.1.2
	- promise:           2.3
	- protobuf:          3.19.4
	- psutil:            5.9.1
	- pyasn1:            0.4.8
	- pyasn1-modules:    0.2.8
	- pydeprecate:       0.3.2
	- pygments:          2.13.0
	- pyparsing:         3.0.9
	- python-dateutil:   2.8.2
	- pytorch-lightning: 1.7.1
	- pyyaml:            6.0
	- requests:          2.28.1
	- requests-oauthlib: 1.3.1
	- rich:              12.5.1
	- rsa:               4.9
	- sentry-sdk:        1.9.4
	- setproctitle:      1.3.2
	- setuptools:        61.2.0
	- shortuuid:         1.0.9
	- six:               1.16.0
	- smmap:             5.0.0
	- tensorboard:       2.10.0
	- tensorboard-data-server: 0.6.1
	- tensorboard-plugin-wit: 1.8.1
	- torch:             1.12.1
	- torchmetrics:      0.9.3
	- torchtext:         0.13.1
	- tqdm:              4.64.0
	- typing-extensions: 4.3.0
	- urllib3:           1.26.11
	- wandb:             0.13.1
	- werkzeug:          2.2.2
	- wheel:             0.37.1
	- yarl:              1.8.1
	- zipp:              3.8.1
* System:
	- OS:                Darwin
	- architecture:
		- 64bit
		-
	- processor:         arm
	- python:            3.9.12
	- version:           Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:22 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T6000
</details>

**Comment by tshu-w** [comment]
When user are not using TensorBoardLogger, `trainer.log_dir` is `logger.save_dir` and we should make sure it's not None.
https://github.com/Lightning-AI/lightning/blob/2622989b108368813fdd3850b002fd1ad69988c2/src/pytorch_lightning/cli.py#L209-L210

**Comment by manangoel99** [comment]
Hi! Not sure why this error is coming back. Looking into it! Meanwhile initialising your CLI with `save_config_callback=None` should work.
 
```
import os

from pytorch_lightning.loggers import WandbLogger
from pytorch_lightning.cli import LightningCLI
from pytorch_lightning import Trainer

from pytorch_lightning.demos.boring_classes import DemoModel, BoringDataModule

def run():
    logger = WandbLogger(project="bug_report")
    cli = LightningCLI(
        DemoModel,
        BoringDataModule,
        trainer_defaults={"logger": logger, "max_epochs": 2, "default_root_dir": os.getcwd()},
        save_config_callback=None
    )


if __name__ == "__main__":
    run()
```

**Comment by tshu-w** [comment]
> Hi! Not sure why this error is coming back. Looking into it! Meanwhile initialising your CLI with `save_config_callback=None` should work.
> 
> ```
> import os
> 
> from pytorch_lightning.loggers import WandbLogger
> from pytorch_lightning.cli import LightningCLI
> from pytorch_lightning import Trainer
> 
> from pytorch_lightning.demos.boring_classes import DemoModel, BoringDataModule
> 
> def run():
>     logger = WandbLogger(project="bug_report")
>     cli = LightningCLI(
>         DemoModel,
>         BoringDataModule,
>         trainer_defaults={"logger": logger, "max_epochs": 2, "default_root_dir": os.getcwd()},
>         save_config_callback=None
>     )
> 
> 
> if __name__ == "__main__":
>     run()
> ```

Yes, I know, I can also init WandbLogger with `save_dir="."` 😄

**Comment by awaelchli** [comment]
This is a duplicate of #12748 and will be solved by #14188. Unless you have any objections, I would close this.
